packages/react-devtools-shared/src/hydration.js

```diff
@@ -295,7 +295,21 @@ export function dehydrate(
         name: data.toString(),
         type,
       };
-
+    case 'window':
+      try {
+        // eslint-disable-next-line no-unused-expressions
+        data.origin;
+      } catch {
+        cleaned.push(path);
+        return {
+          inspectable: false,
+          preview_short: 'Window',
+          preview_long: 'Window',
+          name: 'Window',
+          type,
+        };
+      }
+    // eslint-disable-next-line no-fallthrough
```

Comment:
I don't think we should use the fall through here. Even with the explicit lint rule, it's hard to read and would be easy to break.

Why shouldn't we return the window object regardless of the try/catch?

If we do need the current behavior for a reason I'm not seeing, can we rewrite to make it more explicit, e.g.
```js
    case 'object':
    case 'window':
      if (type === 'window') {
        try {
          // eslint-disable-next-line no-unused-expressions
          data.origin;
        } catch {
          cleaned.push(path);
          return {
            inspectable: false,
            preview_short: 'Window',
            preview_long: 'Window',
            name: 'Window',
            type,
          };
        }
      }

      isPathAllowedCheck = isPathAllowed(path);
      if (level >= LEVEL_THRESHOLD && !isPathAllowedCheck) {
        return createDehydrated(type, true, data, cleaned, path);
      } else {
        const object = {};
        getAllEnumerableKeys(data).forEach(key => {
          const name = key.toString();
          object[name] = dehydrate(
            data[key],
            cleaned,
            unserializable,
            path.concat([name]),
            isPathAllowed,
            isPathAllowedCheck ? 1 : level + 1,
          );
        });
        return object;
      }
```

Reply:
> Why shouldn't we return the window object regardless of the try/catch?

I want to keep the behaviour we had before. We used to display all the elements of the Window object when we could access it.